backend/routes/content.py

```
@@ -710,6 +707,41 @@ def get_bonus_challenges():
+        if user.gems < GEMS_FOR_BONUS_CHALLENGE:
+            logger.error(f"Insufficient gems to buy bonus challenge for user {user_id}")
+            return jsonify({"error": "Insufficient gems to buy bonus challenge"}), 400
+
+        user.gems -= GEMS_FOR_BONUS_CHALLENGE
+        user_module.open = True
+        db.session.add(user_module)
+        db.session.commit()
+
+        logger.info(f"Bonus challenge {challenge_id} unlocked successfully")
+        return jsonify({"message": "Bonus challenge unlocked successfully"}), 200
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)